docs/PLUGIN.md

```diff
@@ -1,19 +1,195 @@
 # Plugin Architecture
 
-Where these application need to interact with external services, the Gateway
+Where these new application need to interact with external services, the Gateway
 code for those services will be in custom C# assemblies with a facade that
 implements a shared interface. The custom assembly will utilize C# [plugin
 support](
 https://learn.microsoft.com/en-us/dotnet/core/tutorials/creating-app-with-plugin-support)
 so that different plugins can be supplied at runtime without having to recompile
 the application.
 
-Examples of assemblies that should be plugin aware:
+Examples of assemblies that could be plugin aware:
 
-* Backend support for different data stores (PostgreSQL, Microsoft SQL Server, Opensearch, etc.)
+* Backend support for different data stores (PostgreSQL, Microsoft SQL Server,
+  Opensearch, etc.)
 * Identity Provider token mapper and client management
   [gateway](https://martinfowler.com/articles/gateway-pattern.html).
 
-## Plugin Detection
+> [!NOTE]
+> As [described below](#frontend-architecture), a plugin architecture for
+> frontend development might not be feasible.
 
-TBD
+## Hypothetical Example on AWS
+
+The Ed-Fi Alliance's first priority in Project Tanager is to build a reference
+architecture using Docker containers that can run in a local development
+environment, in an "on-premises" data warehouse, or in cloud-managed Docker
+services. However, the purpose of this plugin architecture is to make it easy to
+optimize for cloud provider managed services, which may offer better performance and cost compared to other solutions.
+
+For example, one could optimize the Data Management Service solution on AWS:
+
+* Run as an AWS Lambda function (frontend plugin),
+* Connecting to Aurora in PostgreSQL mode for write operations and for reading a
+  specific object,
+* Utilize Kinesis Data Streams to monitor the Aurora database and write to
+  OpenSearch via a Lambda function, and
+* Using OpenSearch for "get all" and "get by query" requests.
+
+```mermaid
+flowchart LR
+
+    classDef plugin fill:#0f0
+
+    subgraph Data Management Service
+        Controller["λ (API Controller)"]:::plugin
+        DataAccess-PG[PostgreSQL Backend]:::plugin
+        DataAccess-OS[OpenSearch Backend]:::plugin
+
+
+        subgraph DMS Core Functionality
+          Valid{Valid Token}
+          OAuth2
+          Write{Write}
+          ReadOne{Read /id}
+        end
+    end
+
+    subgraph Amazon Web Services
+      Aurora[(Aurora)]
+      OpenSearch[(OpenSearch)]
+
+      CDC[Kinesis Data Stream] --> Aurora
```

Comment:
I'd switch the arrow to point from Aurora to Kinesis

Reply:
To be honest I'm not sure if it is a push or a pull. I can switch it.

Reply:
```suggestion
      Aurora --> CDC[Kinesis Data Stream]
```